app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,85 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+  before_action :convert_hyphen_to_dash, only: [:create, :update]
+
+  def index
+    render json: tool_groups_ordered_by_name, include: params[:include], fields: field_params, status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def create_tool
+    tool_group_id = params[:tool_group_id].to_i
+    resource_id = params.dig(:data, :relationships, :tool, :data, :id).to_i
+    suggestions_weight = params.dig(:data, :attributes, "suggestions-weight").to_f
+
+    resource_tool_group = ResourceToolGroup.create!(
+      tool_group_id: tool_group_id,
+      resource_id: resource_id,
+      suggestions_weight: suggestions_weight
+    )
+
+    response.headers["Location"] = "tool-groups/#{resource_tool_group.id}"
+    render json: resource_tool_group, status: :created
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {errors: formatted_errors(e)}, status: :unprocessable_entity
+  end
+
+  def update_tool
+    existing = resource_tool_group
+    existing.update!(
+      resource_id: params[:data][:attributes]["resource-id"],
+      suggestions_weight: params[:data][:attributes]["suggestions-weight"]
+    )
+    render json: existing, status: :accepted
+  end
+
+  def delete_tool
+    resource_tool_group.destroy!
+    head :no_content
+  end
+
+  def show
+    render json: load_tool_group, include: params[:include], fields: field_params, status: :ok
+  end
+
+  def destroy
+    tool_group = ToolGroup.find(params[:id])
+    tool_group.destroy!
```

Comment:
> ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  update or delete on table "tool_groups" violates foreign key constraint "fk_rails_c56fad9d31" on table "resource_tool_groups"
DETAIL:  Key (id)=(4) is still referenced from table "resource_

https://app.rollbar.com/a/Cru/fix/item/mobile-content-api/323

<img width="770" alt="Screenshot 2023-08-02 at 2 29 18 PM" src="https://github.com/CruGlobal/mobile-content-api/assets/47818/89aef078-36ad-4a97-96fc-6a03dcd9dd40">

Reply:
deleting a tool-group that still contains relationships to `resources` should succeed and delete those relationships.

it should NOT cascade and delete the `resources` themselves, just the relationships

Reply:
should be fixed now